build_functions/CMakeLists.txt

```diff
@@ -193,3 +193,43 @@ function(compileTargetAsC11 theTarget)
     set_target_properties(${theTarget} PROPERTIES CXX_STANDARD 11)
   endif()
 endfunction()
+
+# variable to store list of libs that must be checked for reals
+set(LIBS_TO_BE_REAL_CHECKED CACHE INTERNAL "LIBS_TO_BE_CHECKED")
+
+# overloading the add_library function to store name of library if it meets criteria
+function (add_library())
+    _add_library(${ARGV})
```

Comment:
>_add_library [](http://example.com/codeflow?start=4&length=12)

this overload is not a supported feature of CMake. Would be nice to have a link to the original place where you've picked up this trick